Storage/SQL/SQLiteBookmarksBase.swift

```diff
@@ -35,4 +45,21 @@ public class SQLiteBookmarks: BookmarksModelFactorySource {
         
         return self.db.queryReturnsResults(sql, args: args)
     }
+
+    public func getURLForKeywordSearch(keyword: String) -> Deferred<Maybe<String>> {
+        let sql = "SELECT bmkUri FROM \(TableBookmarksBuffer) WHERE " +
```

Comment:
I don't think querying the buffer is the correct choice here.

The buffer will — as soon as bidirectional syncing is enabled — be empty. That this code works now is almost an accident!

Because we don't allow you to edit bookmarks locally, or add keywords to them locally, an acceptable short-term solution here is to query the overlay of remote bookmarks on top of merged bookmarks (`ViewBoomarksBufferOnMirror`). A longer-term solution is to query everything: use `ViewAllBookmarks`.

Those views don't pass through `keyword` (because we haven't used it so far), so you will need a small database migration to do what you need. Ask @sleroux for help with that.